src/Redux/Missions/Missions.js

```diff
@@ -20,24 +20,38 @@ export default function missionReducer(state = initialState, action) {
   switch (action.type) {
     case `${FETCH_MISSIONS}/fulfilled`:
       return [...action.payload];
-    case JOIN_MISSION:
-      return state;
-    case LEAVE_MISSION:
-      return state;
+    case TOGGLE_JOIN_MISSION:
+      return (state.map((missions) => {
+        if (missions.mission_id === action.mission_id) {
+          return {
+            ...missions,
+            reserved: !missions.reserved,
+          };
+        }
+        return missions;
+      }));
     default:
       return state;
   }
 }
 
 // Action Creators
+const addJoined = (array) => {
+  const missions = array.map((obj) => ({ ...obj, reserved: false }));
+  return missions;
```

Comment:
I'll suggest changing the key from 'reserved' to 'joined' as that better describes the action.
`({ ...obj, joined: false })`
This will be better.